addons/product/models/product.py

```diff
@@ -561,6 +561,48 @@ def view_header_get(self, view_id, view_type):
             return _('Products: ') + self.env['product.category'].browse(self._context['categ_id']).name
         return res
 
+    # returns the checksum of the ean13, or -1 if the ean has not the correct length, ean must be a string
+    def ean13_checksum(self, ean):
+        code = list(ean)
+        if len(code) != 13:
+            return -1
+        return self._ean_checksum(code)
+
+    # returns the checksum of the ean8, or -1 if the ean has not the correct length, ean must be a string
+    def ean8_checksum(self, ean):
+        code = list(ean)
+        if len(code) != 8:
+            return -1
+        return self._ean_checksum(code)
+
+    def _ean_checksum(self, code):
+        oddsum = evensum = 0
+        code = code[:-1] # Remove checksum
+        index = len(code) -1
+        even = False
+        while index > -1:  # The checksum on a barcode is calculated from right to left
+            if even:
+                evensum += int(code[index])
+            else:
+                oddsum += int(code[index])
+            even = not even
+            index -= 1
+        total = 3 * oddsum + evensum
+        return int((10 - total % 10) % 10)
```

Comment:
```suggestion
    @api.model
    def get_barcode_check_digit(self, numeric_barcode):
        """ Computes and returns the barcode check digit. The used algorithm
        follows the GTIN specifications and can be used by all compatible
        barcode nomenclature, like as EAN-8, EAN-12 (UPC-A) or EAN-13.

        https://www.gs1.org/sites/default/files/docs/barcodes/GS1_General_Specifications.pdf
        https://www.gs1.org/services/how-calculate-check-digit-manually

        :param numeric_barcode: the barcode to verify/recompute the check digit
        :type numeric_barcode: str
        :return: the number corresponding to the right check digit
        :rtype: int
        """
        # Multiply value of each position by
        # N1  N2  N3  N4  N5  N6  N7  N8  N9  N10 N11 N12 N13 N14 N15 N16 N17 N18
        # x3  X1  x3  x1  x3  x1  x3  x1  x3  x1  x3  x1  x3  x1  x3  x1  x3  CHECKSUM
        oddsum = evensum = total = 0
        code = numeric_barcode[-2::-1]  # Remove the check digit and reverse the barcode.
        # The CHECKSUM digit is removed because it will be recomputed and it must not interfer with
        # the computation. Also, the barcode is inverted, so the barcode length doesn't matter.
        # Otherwise, the digits' group (even or odd) could be different according to the barcode length.
        for i, digit in enumerate(code):
            if i % 2 == 0:
                evensum += int(digit)
            else:
                oddsum += int(digit)
        total = evensum * 3 + oddsum
        return (10 - total % 10) % 10
```
(I forgot to add this comment to the review before to submit it...)